src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+    private float threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > threshold) {
+            resize();
+        }
+        int hash = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        if (table[hash] == null) {
+            table[hash] = newNode;
+            size++;
+            return;
+        }
 
+        Node<K, V> currentNode = table[hash];
+        while (currentNode.next != null && !Objects.equals(key,currentNode.key)) {
+            currentNode = currentNode.next;
+        }
+        if (Objects.equals(key, currentNode.key)) {
+            currentNode.value = value;
+            return;
+        }
+        currentNode.next = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[getIndex(key)];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        final int GrowFactor = 2;
```

Comment:
[This should be a constant value](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)